### PR TITLE
Better api

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: lib/prestige/presto_client/request_stream.ex [hunk 1]
```diff
@@ -0,0 +1,109 @@
+defmodule Prestige.PrestoClient.RequestStream do
+  @moduledoc false
+  use Tesla
```
Comment (jeffgrunewald) [inline lib/prestige/presto_client/request_stream.ex]: I thought you were going with mojito b/c it was lighter weight.

File: lib/prestige.ex [hunk 2]
```diff
@@ -3,6 +3,9 @@ defmodule Prestige do
   An elixir client for [Prestodb](http://prestodb.github.io/).
   """
 
+  alias Prestige.PrestoClient
```
Comment (jeffgrunewald) [inline lib/prestige.ex]: Why not just `Client` since it’s in the context of Presto already?

File: lib/prestige/presto_client.ex [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  def close_statement(session, name) do
+    deallocate_statement = "DEALLOCATE PREPARE #{name}"
+
+    request = %Request{
+      session: session,
+      statement: deallocate_statement,
+      prepare_statement: false
+    }
+
+    [_result] =
```
Comment (jeffgrunewald) [inline lib/prestige/presto_client.ex]: Do you really need this match against an unused result if you’re not handling when the result _doesn’t_ come back as a list?

File: lib/prestige/presto_client/request_stream.ex [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  alias Prestige.PrestoClient.Request
+  alias Prestige.PrestoClient.Arguments
```
Comment (jeffgrunewald) [inline lib/prestige/presto_client/request_stream.ex]: Compress these two aliases

File: mix.exs [hunk 5]
```diff
@@ -32,7 +33,8 @@ defmodule Prestige.MixProject do
       {:temporary_env, "~> 2.0", only: :test},
       {:credo, "~> 1.1", only: [:dev, :test], runtime: false},
       {:ex_doc, "~> 0.19", only: :dev},
-      {:husky, "~> 1.0", only: :dev, runtime: false}
+      {:husky, "~> 1.0", only: :dev, runtime: false},
```
Comment (jeffgrunewald) [inline mix.exs]: I think we’re pretty committed to removing husky at this point.

File: lib/prestige/presto_client/request_stream.ex (hunk 1 above)
Comment (bbalser) [reply lib/prestige/presto_client/request_stream.ex]: I still think I want to, but I pushed that to a future concern.

File: test/integration/prestige_test.exs [hunk 6]
```diff
@@ -0,0 +1,105 @@
+defmodule PrestigeTest do
+  use Prestige.Case
```
Comment (bbrewer-papa-x) [inline test/integration/prestige_test.exs]: Really liking this

File: lib/prestige/result.ex [hunk 7]
```diff
@@ -1,87 +1,56 @@
...
-  end
+  alias Prestige.ColumnDefinition
 
-  def transform({:ok, %Tesla.Env{status: 200, body: body}}, rows_as_maps) do
-    data =
-      body
-      |> Map.get("data", [])
-      |> Enum.map(&transform_row(&1, Map.get(body, "columns", []), rows_as_maps))
+  @type t :: %Prestige.Result{
+          columns: list(ColumnDefinition.t()),
```
Comment (bbrewer-papa-x) [inline lib/prestige/result.ex]: Thank you for doing this. Going to be very useful for multi-query in https://github.com/smartcitiesdata/discovery_api

**Pull request merged**